server/src/main/java/org/apache/druid/indexing/overlord/supervisor/Supervisor.java

```diff
@@ -93,4 +94,10 @@ default Boolean isHealthy()
   LagStats computeLagStats();
 
   int getActiveTaskGroupsCount();
+
+  /** Handoff the task group with id=taskGroupId the next time the supervisor runs regardless of task run time*/
+  default void handoffTaskGroupsEarly(List<Integer> taskGroupIds)
+  {
+    throw new NotImplementedException("Supervisor does not have the feature to handoff task groups early implemented");
```

Comment:
Nit: Why not use `UnsupportedOperationException` instead?
The error message seems to suggest that this is more of an "unsupported" scenario.